_posts/2022-08-18-Day9.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: "Creating C++ Projects with Multiple Files"
+author: Joshua Blomgren
+---
+
+## Multiple Files??
+
+Because C++ involves compiling and linking code into one program, it is actually quite simple to write multiple files for a project and access functions that are located in other files. This is done through declarations, a.k.a declaring in one file that a certain function exists in another file. Function definitions can only be written once, but there is no limit to function declarations. A function declaration is identitical to the first line of a function definition *except* that there’s a semicolon at the end.
+```
+ int example_function(std::string s, int i); 
+```
+
+Because the compiler reads top down, it is a good idea to declare functions exist before they are used in your code. Declaring functions before they are used in main is known as **forward declaration**. 
+
+However, it would be very tedious and messy to have forward declarations for every file that you write. Good news! It’s possible to put all of them into one file and access them using #include. These files are called **header files** that have the extension .hpp (or .hxx, .h, etc). While external libraries like iostream and vector are included using < >, internal files of the same project, like headers, are included using “ “. 
```

Comment:
I think it is good for OOP. I read a lot about C++ is better in OOP than Python, we might be able to dig in more during the semester.